javalib/src/main/scala/java/net/SocketHelpers.scala

```diff
@@ -248,4 +258,98 @@ object SocketHelpers {
         if (status == 0) Some(fromCString(host)) else None
       }
     }
+
+  // True if at least one non-loopback interface has an IPv6 address.
+  private def isIPv6Configured(): Boolean = {
+    if (isWindows) {
+      false // Support for IPv6 is neither implemented nor tested.
+    } else {
+      /* The lookup can not be a local address. This one of two IPv6
+       * addresses for the famous, in the IPv6 world, www.kame.net
+       * IPv6 dancing kame (turtle). The url from Ipv6 for fun some time
+       */
+      val kameIPv6Addr = c"2001:2F0:0:8800:0:0:1:1"
+
+      val hints = stackalloc[addrinfo]() // stackalloc clears its memory
+      val ret = stackalloc[Ptr[addrinfo]]()
+
+      hints.ai_family = AF_INET6
+      hints.ai_flags = AI_NUMERICHOST | AI_ADDRCONFIG
+      hints.ai_socktype = SOCK_STREAM
+
+      val gaiStatus = getaddrinfo(kameIPv6Addr, null, hints, ret)
+      val result =
+        if (gaiStatus != 0) {
+          false
+        } else {
+          try {
+            val ai = !ret
+            if ((ai == null) || (ai.ai_addr == null)) {
+              false
+            } else {
+              ai.ai_addr.sa_family == AF_INET6.toUShort
+            }
+          } finally {
+            freeaddrinfo(!ret)
+          }
+        }
+
+      result
+    }
+  }
+
+  // A Single Point of Truth to toggle IPv4/IPv6 underlying transport protocol.
+  private lazy val preferIPv4Stack: Boolean = {
+    val prop = System.getProperty("java.net.preferIPv4Stack")
+    // Java defaults to "false", did System properties override?
+    val forceIPv4 = ((prop != null) && (prop.toLowerCase() == "true"))
+    forceIPv4 || !isIPv6Configured() // Do the expensive test last.
+  }
+
+  private[net] def getPreferIPv4Stack(): Boolean = preferIPv4Stack
```

Comment:
I think you could probably just use the lazy vals directly here and the ones below.

Reply:
Eric,  Thank you for the review and suggestion.

I had considered what you suggested and decided that I wanted 'getters'.  At this point in 
development I am not sure enough of the implementation of the getters to collapse them.
I _think_ they will stay the same, but I have a fear-in-the-night that they will need change/tuning.
I want to be able to do it in one place and not have to chase all over `java.net`.

Reply:
> This looks really good - the documentation too. I am not expert enough in this area for a real review.

Thank you for looking and for the encouragement, now & along the way.
 
If you look carefully enough, the documentation (and the code) has your influence all over it.

I am working on a Windows issue that cropped up in this last WIP commit.   Arman talked
in Discourse and recommended an idiom for "if include file not found".  I thought I had
memorized it, but now that I need it, all I have are holes in my memory. Oops! Did I say
that in public?